projects/approved-premises-and-delius/src/integrationTest/kotlin/uk/gov/justice/digital/hmpps/DocIntegrationTest.kt

```diff
@@ -5,29 +5,35 @@ import org.junit.jupiter.api.Test
 import org.springframework.beans.factory.annotation.Autowired
 import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc
 import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.http.MediaType
 import org.springframework.test.web.servlet.MockMvc
-import org.springframework.test.web.servlet.MvcResult
-import org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get
+import org.springframework.test.web.servlet.get
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders.asyncDispatch
 import org.springframework.test.web.servlet.result.MockMvcResultMatchers.*
 import org.springframework.util.ResourceUtils
 import uk.gov.justice.digital.hmpps.data.generator.PersonGenerator
 import uk.gov.justice.digital.hmpps.test.MockMvcExtensions.withToken
 
 @AutoConfigureMockMvc
 @SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
-internal class DocIntegrationTest {
-    @Autowired
-    lateinit var mockMvc: MockMvc
+internal class DocIntegrationTest(
+    @Autowired private val mockMvc: MockMvc
+) {
 
     @Test
     fun `document is downloaded`() {
-        mockMvc
-            .perform(
-                get("/documents/A000001/00000000-0000-0000-0000-000000000001").accept("application/octet-stream")
-                    .withToken()
-            )
-            .andExpect(request().asyncStarted())
-            .andDo(MvcResult::getAsyncResult)
+        // 1. Trigger async request using Kotlin DSL
+        val mvcResult = mockMvc.get("/documents/A000001/00000000-0000-0000-0000-000000000001") {
+            withToken()
+            accept = MediaType.APPLICATION_OCTET_STREAM
+        }
+            .andExpect {
+                request { asyncStarted() }
+            }
+            .andReturn()
+
+        // 2. Continue async processing using perform + asyncDispatch (DSL not supported here)
+        mockMvc.perform(asyncDispatch(mvcResult))
```

Comment:
Could this be done in one go?  E.g.

```suggestion
        mockMvc
            .get("/documents/A000001/00000000-0000-0000-0000-000000000001") {
                withToken()
                accept = MediaType.APPLICATION_OCTET_STREAM
            }
            .andExpect { request { asyncStarted() } }
            .asyncDispatch()
```